cognite_toolkit/_cdf_tk/utils/cdf.py

```diff
@@ -143,20 +143,33 @@ def read_auth(
 
 
 def metadata_key_counts(
-    client: ToolkitClient, resource: Literal["assets", "events", "files", "timeseries", "sequences"]
-) -> list[dict[str, int | str]]:
+    client: ToolkitClient,
+    resource: Literal["assets", "events", "files", "timeseries", "sequences"],
+    data_sets: list[int] | None = None,
+    hierarchies: list[int] | None = None,
+) -> list[tuple[str, int]]:
     """Get the metadata key counts for a given resource.
 
     Args:
         client: ToolkitClient instance
         resource: The resource to get the metadata key counts for. Can be one of "assets", "events", "files", "timeseries", or "sequences".
+        data_sets: A list of data set IDs to filter by. If None, no filtering is applied.
+        hierarchies: A list of hierarchy IDs to filter by. If None, no filtering is applied.
 
     Returns:
         A dictionary with the metadata keys as keys and the counts as values.
     """
+    where_clause = ""
+    if data_sets is not None and hierarchies is not None:
+        where_clause = f"\n         WHERE dataSetId IN ({''.join(map(str, data_sets))}) AND rootId IN ({','.join(map(str, hierarchies))})"
```

Comment:
bug in joining data_sets, empty string used as separator

Reply:
would it make sense to avoid repeating the conditions (and allowing subtle differences) by doing something like
```
data_sets_condition = f"dataSetId IN ({','.join(map(str, data_sets))})" if data_sets is not None else None
hierarchies_condition = f"rootId IN ({','.join(map(str, hierarchies))})" if hierarchies is not None else None
conditions = [c for c in [data_sets_condition,hierarchies_condition] if c is not None]
where_clause = f"\n     WHERE {' AND '.join(conditions)}" if conditions else ""
```

Reply:
I chose to make it more explicit and avoid the above as it is harder to read. So I prefer keeping it as is.